src/util/bip152.rs

```diff
@@ -0,0 +1,438 @@
+// Rust Bitcoin Library
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! BIP152 Compact Blocks
+//!
+//! Implementation of compact blocks data structure and algorithms.
+//!
+
+use std::{convert, error, fmt, io, mem};
+
+use hashes::{hex, sha256, siphash24, Hash};
+
+use hash_types::BlockHash;
+use blockdata::block::{Block, BlockHeader};
+use blockdata::transaction::Transaction;
+use consensus::encode::{self, Decodable, Encodable, VarInt};
+use util::endian;
+
+/// A BIP-152 error
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum Error {
+    /// An unknown version number was used.
+    UnknownVersion,
+    /// A transaction index is requested that is out
+    /// of range from the corresponding block.
+    TxIndexOutOfRange(u64),
+    /// The prefill slice provided was invalid.
+    InvalidPrefill,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::UnknownVersion => write!(f, "an unknown version number was used"),
+            Error::TxIndexOutOfRange(i) => write!(f, "a transaction index is requested that is \
+                out of range from the corresponding block: {}", i,
+            ),
+            Error::InvalidPrefill => write!(f, "the prefill slice provided was invalid"),
+        }
+    }
+}
+
+impl error::Error for Error {
+    fn cause(&self) -> Option<&error::Error> {
+        None
+    }
+
+    fn description(&self) -> &str {
+        "description() is deprecated; use Display"
+    }
```

Comment:
Are either of these method implementations necessary? 

Nowhere else do we implement `Error::description` and `Error::cause` [defaults to `None`](https://doc.rust-lang.org/src/std/error.rs.html#151).

Reply:
No, they were needed when our MSRV was 1.22 (when this PR was opened :))